gallery_dl/extractor/tiktok.py

```diff
@@ -0,0 +1,207 @@
+                    url = img["imageURL"]["urlList"][0]
+                    name_and_ext = text.nameext_from_url(url)
+                    yield Message.Url, url, {
+                        "title"     : title,
+                        "date"      : date,
+                        "id"        : id,
+                        "index"     : i + 1,
+                        "img_id"    : name_and_ext["filename"].split("~")[0],
+                        "extension" : name_and_ext["extension"],
+                        "width"     : img["imageWidth"],
+                        "height"    : img["imageHeight"]
+                    }
+            elif videos:
+                if len(original_title) == 0:
+                    title = "TikTok video #{}".format(id)
+                    title = title[:150]
+            else:
+                self.log.info("Skipping video post %s", tiktok_url)
+            if videos:
+                yield Message.Url, "ytdl:" + tiktok_url_to_use, {
+                    "filename"  : "",
+                    "extension" : "",
+                    "title"     : title,
+                    "date"      : date,
+                    "id"        : id,
+                    "index"     : "",
+                    "img_id"    : ""
+                }
+
+
+class TiktokPostExtractor(TiktokExtractor):
+    """Extract a single video or photo TikTok link"""
+
+    subcategory = "post"
+    pattern = POST_PATTERN
+    example = "https://www.tiktok.com/@chillezy/photo/7240568259186019630"
+
+
+class TiktokVmpostExtractor(TiktokExtractor):
+    """Extract a single video or photo TikTok VM link"""
+
+    subcategory = "vmpost"
+    pattern = VM_POST_PATTERN
+    example = "https://vm.tiktok.com/ZGdh4WUhr/"
+
+
+class TiktokSharepostExtractor(TiktokExtractor):
+    """Extract a single video or photo TikTok share link"""
+
+    subcategory = "sharepost"
+    pattern = SHARE_PATTERN
+    example = "https://www.tiktokv.com/share/video/7240568259186019630"
+
+
+class TiktokUserExtractor(TiktokExtractor):
+    """Extract a TikTok user's profile"""
+
+    subcategory = "user"
+    pattern = USER_PATTERN + r"/*$"
+    example = "https://www.tiktok.com/@chillezy"
+
+    def urls(self):
+        """Attempt to use yt-dlp/youtube-dl to extract links from a
+        user's page"""
+
+        try:
+            module = ytdl.import_module(self.config("module"))
+        except (ImportError, SyntaxError) as exc:
+            self.log.error("Cannot import module '%s'",
+                           getattr(exc, "name", ""))
+            self.log.debug("", exc_info=exc)
+            raise exception.ExtractionError("yt-dlp or youtube-dl is required "
+                                            "for this feature!")
+        with ytdl.construct_YoutubeDL(
+            module=module,
+            obj=self,
+            user_opts={
+                "ignore_no_formats_error": True,
+                "cookiefile": self.cookies_file,
```

Comment:
There is probably some way to transfer gallery-dl cookies (`self.cookies`) to yt-dlp directly. The `ytdl` downloader has a [`forward-cookies`](https://gdl-org.github.io/docs/configuration.html#downloader-ytdl-forward-cookies) option for this.